dependencies.yaml

```diff
@@ -258,10 +258,10 @@ dependencies:
       - output_types: [conda, requirements, pyproject]
         packages:
           - click
-          - lazy_loader>=0.1
-          - numpy>=1.23.4,<3.0a0
-          - scikit-image>=0.19.0,<0.25.0a0
-          - scipy>=1.6.0
+          - lazy-loader>=0.4
```

Comment:
Since `-` and `_` are distinct in conda and not in pip, we should double check this naming is matched by the conda package

Otherwise we may want to split conda and pip into two different sections

Reply:
`lazy-loader` is the new name, and this should be correct for both PyPI and conda-forge.

See: https://github.com/conda-forge/lazy_loader-feedstock/pull/11

Also https://pypi.org/project/lazy_loader/ redirects to https://pypi.org/project/lazy-loader/